integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py

```diff
@@ -307,5 +333,21 @@ def run(self, messages: List[ChatMessage]):
                             name=part.function_call.name,
                         )
                     )
+        return replies
 
-        return {"replies": replies}
+    def get_stream_response(self, stream, streaming_callback: Callable[[StreamingChunk], None]) -> List[ChatMessage]:
+        """
+        Extracts the responses from the Google AI streaming response.
+
+        :param stream: The streaming response from the Google AI request.
+        :param streaming_callback: The handler for the streaming response.
+        :returns: The extracted response with the content of all streaming chunks.
+        """
+        responses = []
+        for chunk in stream:
+            content = chunk.text if len(chunk.parts) > 0 and "text" in chunk.parts[0] else ""
+            streaming_callback(StreamingChunk(content=content, meta=chunk.to_dict()))
+            responses.append(content)
+
+        combined_response = "".join(responses).lstrip()
+        return [ChatMessage.from_system(content=combined_response)]
```

Comment:
**Unrelated but relevant**

What I'm noticing here and also in the existing code is that we are using the `system` role, while the responses we get from the model should have the `assistant` role. Could you investigate?

In case we are doing it wrong, please open an issue and then we can work on a fix in another PR.

Reply:
Thanks for pointing this out. I cross checked in other integrations and the messages from model are marked as `assistant`. I'll update this PR.

Reply:
Thanks! I suggest doing this in another PR, also because PR titles are used to populate the CHANGELOG.md file.